src/casp/services/api/services/cell_operations_service.py

```diff
@@ -139,17 +114,17 @@ def __validate_knn_response(embeddings: np.array, knn_response: MatchResult) ->
                 raise exceptions.VectorSearchResponseError("Vector Search returned a match with 0 neighbors.")
 
     async def __get_knn_matches_from_embeddings_with_retry(
-        self, embeddings: np.array, model_name: str, chunk_matches_function: t.Callable
+        self, embeddings: np.array, model: models.CASModel, chunk_matches_function: t.Callable
     ) -> MatchResult:
         """
         Get KNN matches for embeddings broken into chunks with retry logic
 
         :param embeddings: Embeddings to match.
-        :param model_name: Model name to use for matching.
+        :param model: Model to use for matching.
 
         :return: List of lists of MatchNeighbor objects.
         """
-        index = self.__get_match_index_endpoint_for_model(model_name=model_name)
+        index = model.cas_matching_engine
 
         matching_client = MatchingClient.from_index(index)
```

Comment:
Very small thing, just saving one line, but we can do this as well (and thus removing `index = model.cas_matching_engine`):
```suggestion
        matching_client = MatchingClient.from_index(model.cas_matching_engine)
```